flask_appbuilder/hooks.py

```diff
@@ -0,0 +1,76 @@
+def before_request(*args, **kwargs):
+    """
+        Use this decorator to enqueue methods to be invoked
+        before each handler in the view. If method returns
+        a value other than :code:`None`, then that value
+        will be returned to the client. If invoked with the
+        :code:`only` kwarg, the hook will only be invoked for
+        the given list of handler methods.
+
+        Examples::
+
+            class MyFeature(ModelView)
+
+                @before_request
+                def ensure_feature_is_enabled(self):
+                    if self.feature_is_disabled:
+                        return self.response_404()
+                    return None
+
+                # etc...
+
+
+            class MyView(ModelRestAPI):
+
+                @before_request(only=["create", "update", "delete"])
+                def ensure_write_mode_enabled(self):
+                    if self.read_only:
+                        return self.response_400()
+                    return None
+
+                # etc...
+
+
+        :param kwargs:
+            The :code:`only` kwarg scopes the method being decorated
+            to run only for the given list of handler methods. If
+            abset, the method will be invoked before all handlers.
+    """
+    only = kwargs.get("only", None)
```

Comment:
Why not make the signature like `before_request(only: Optional[List[str]]) -> Callable[..., Any]:`

Reply:
It's not as simple as that since it can be used as a decorator without being invoked by user (in that case, `args[0]` is a bound function) or it can be invoked by user with `only` kwarg which returns a function that takes a bound function.

I'll see what I can do for this.

Reply:
👍🏻